test/test_datasets.py

```diff
@@ -280,6 +283,92 @@ def test_ucf101(self):
                         self.assertEqual(audio.numel(), 0)
                         self.assertEqual(label, 1)
 
+    def test_places365(self):
+        for split, small in itertools.product(("train-standard", "train-challenge", "val", "test"), (False, True)):
+            with places365_root(split=split, small=small) as places365:
+                root, data = places365
+
+                dataset = torchvision.datasets.Places365(root, split=split, small=small, download=True)
+                self.generic_classification_dataset_test(dataset, num_images=len(data["imgs"]))
+
+    def test_places365_transforms(self):
+        expected_image = "image"
+        expected_target = "target"
+
+        def transform(image):
+            return expected_image
+
+        def target_transform(target):
+            return expected_target
+
+        with places365_root() as places365:
+            root, data = places365
+
+            dataset = torchvision.datasets.Places365(
+                root, transform=transform, target_transform=target_transform, download=True
+            )
+            actual_image, actual_target = dataset[0]
+
+            self.assertEqual(actual_image, expected_image)
+            self.assertEqual(actual_target, expected_target)
+
+    @mock.patch("torchvision.datasets.utils.download_url")
+    def test_places365_downloadable(self, download_url):
+        for split, small in itertools.product(("train-standard", "train-challenge", "val", "test"), (False, True)):
+            with places365_root(split=split, small=small) as places365:
+                root, data = places365
+
+                torchvision.datasets.Places365(root, split=split, small=small, download=True)
+
+        urls = {call_args[0][0] for call_args in download_url.call_args_list}
+        for url in urls:
+            with self.subTest(url=url):
+                response = urlopen(Request(url, method="HEAD"))
+                assert response.code == 200, f"Server returned status code {response.code} for {url}."
```

Comment:
Interesting approach.

Do you think this could introduce flakiness in the tests if the servers stop working?

Reply:
IMO, if the servers stop working this will exactly do as intended: if it fails, we know the download is not working properly. Before we were dependent on a user reporting this issue.

To take this even further: I think it would be a good idea to do this for all datasets and for example run this on a schedule (for example daily).

Reply:
Yes, that is a good point and we should extend it to all datasets that support download.

I think we should do something like that, but maybe on a separate CI job so that we don't pollute the rest of the tests if one of the dataset URLs stop working.

Thoughts?

Reply:
I'll open an issue to discuss this.